catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java

```diff
@@ -55,19 +55,17 @@ public class MysqlTableOperations extends JdbcTableOperations {
 
   @Override
   public List<String> listTables(String databaseName) throws NoSuchSchemaException {
-    try (Connection connection = getConnection(databaseName)) {
-      try (Statement statement = connection.createStatement()) {
-        String showTablesQuery = "SHOW TABLES";
-        ResultSet resultSet = statement.executeQuery(showTablesQuery);
-        List<String> names = new ArrayList<>();
-        while (resultSet.next()) {
-          String tableName = resultSet.getString(1);
-          names.add(tableName);
+    final List<String> names = Lists.newArrayList();
+
+    try (Connection connection = getConnection(databaseName);
+        ResultSet tables = getTables(connection)) {
+      while (tables.next()) {
+        if (Objects.equals(tables.getString("TABLE_CAT"), databaseName)) {
```

Comment:
tables.getString(“TABLE_SCHEM”) is null for MySQL?

Reply:
Yes, MySQL use TABLE_CAT instead of TABLE_SCHEM

Reply:
```
TABLE_CAT: test_db__65e84ca8
TABLE_SCHEM: null
TABLE_NAME: RETvkaoSYChzfOJd_op_table
TABLE_TYPE: TABLE
REMARKS: test_comment
```

Reply:
got it ,thanks